src/game/default/g_cmd.c

```diff
@@ -757,11 +786,16 @@ _Bool G_AddClientToTeam(g_entity_t *ent, const char *team_name) {
 	if (ent->client->locals.persistent.team == team)
 		return false;
 
+	if (g_level.gameplay == GAME_DUEL && G_TeamSize(team) > 0) {
+		gi.ClientPrint(ent, PRINT_HIGH, "Only 1 player per team allowed in Duel mode\n");
+		return false;
+	}
+	
```

Comment:
Now that I actually understand what you were saying (I replied right after waking up), yes it should kick everyone off teams if g_gameplay is changed to duel. That sanity check was intended to prevent spectator3 from joining a team that already has a player on it in duel mode.